src/adapter/driven/checkout/repository/checkoutRepository.ts

```diff
@@ -0,0 +1,31 @@
+import QRCode from 'qrcode';
+
+import CheckoutRepository, { Pagamento } from "~core/applications/repositories/checkoutRepository";
+import { Pedido, statusDoPedido } from "~core/domain/pedido";
+import PedidoDataBaseRepository from '~driven/infra/repository/pedidoDatabaseRepository';
+
+
+// FAKE CHECKOUT 
+export default class FakeCheckout implements CheckoutRepository {
+  async geraPagamento(metodoDePagamentoId: string, pedido: Pedido): Promise<Pagamento> {
+    // Deve validar a forma de pagamento e enviar o valor do pedido para o pagamento externo
+    // O Fake checkout apenas vai gerar o fake qrcode e automaticamente mudar o status para aguardando preparo
+
+    try {
+      const dbPedidosRepository = new PedidoDataBaseRepository();
+
+      await dbPedidosRepository.atualizaPedido({ id: pedido.id, status: statusDoPedido.AGUARDANDO_PREPARO })
+
```

Comment:
Esse método me parece bem acoplado, pois a classe PedidoDataBaseRepository tá sendo chamada diretamente dentro da função. Se fosse usar o Repository de pedidos acredito que deveria ser um parâmetro no construtor da classe. Porém acho que o repository deveria ter cara bem de crud mesmo (criaCheckout, atualizaChekout...) até fiquei pensando se isso não deveria ser dentro de Fatura 🤔. E seguindo essa lógica de Repository ser bem Crud o método da Service seria geraPagamento que chama o atualizaCheckout ou atualizaFatura e também atualizaPedido. Com a repository estilo CRUD a gente consegue reutilizar os métodos para vários use cases.

Fez sentido?

Reply:
Na implementação sem ser o Fake esse repositório vai chamar apenas o SDK e retornar o QR code, quem vai manipular o pedido seria um endpoint do nosso projeto que o mercado pago chamaria. Uma alternativa seria eu já implementar esse endpoint e chamar nessa classe pra fazer o pagamento fake

Reply:
concordo com o @diorgeneseugenio nesse ponto, talvez fosse melhor já deixar desacoplado mesmo se for só pra avaliação